### PR TITLE
{Release} Upgrade to Azure CLI 2.40.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

File: src/azure-cli-core/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,15 @@
 Release History
 ===============
 
+2.40.0
+++++++
+* `aaz`: Support to use cli_ctx to initiate AAZCommand class (#23459)
```
Comment (evelyn-ys) [inline src/azure-cli-core/HISTORY.rst:8]:
```suggestion
* `aaz`: Support using `cli_ctx` to initiate AAZCommand class (#23459)
```

**Review by jiasli — APPROVED** [review]

File: src/azure-cli-core/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,15 @@
...
+* `aaz`: Fix AAZBearerTokenCredentialPolicy to support MSIAuthenticationWrapper crendital (#23471)
```
Comment (jsntcy) [inline src/azure-cli-core/HISTORY.rst:9]:
```suggestion
* `aaz`: Fix AAZBearerTokenCredentialPolicy to support MSIAuthenticationWrapper credential (#23471)
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,150 @@
...
+* `az cosmos db service`: Add service support for cosmosDB (#23555)
+* `az cosmosdb gremlin graph`: Add analyticalStorageTTL property to sql containers (#23555)
+
+**Feedback**
+
+* `az survey`: New command for CLI survey (#23460)
+
+**Monitor**
+
+* Add new command: `az monitor action-group test-notifications create` (#23411)
```
Comment (necusjz) [inline src/azure-cli/HISTORY.rst:83]:
```suggestion
* `az monitor action-group test-notifications create`: Add new command (#23411)
```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,150 @@
 Release History
 ===============
 
+2.40.0
+++++++
+
+**ACR**
+
+* Connected registry error help command typo (#23431)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:11]:
```suggestion

```

Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:11]: Not easy to understand this note.

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,150 @@
...
+* `az acr config authentication-as-arm show`: Add new command to support showing the configured 'Azure AD authenticate as ARM' policy (#23323)
+* `az acr config authentication-as-arm update`: Add new command to support updating 'Azure AD authenticate as ARM' policy (#23323)
+* `az acr manifest show`: Improve performance (#23523)
+* `az acr config soft-delete show`: Add new command to show soft-delete policy (#22959)
+* `az acr config soft-delete update`: Add new command to update soft-delete policy (#22959)
+* `az acr repository list-deleted`: Add new command to list deleted repositories (#22959)
+* `az acr manifest list-deleted`: Add new command to list deleted manifests (#22959)
+* `az acr manifest list-deleted-tags`: Add new command to list deleted tags (#22959)
+* `az acr manifest restore`: Add new command to restore deleted manifests and tags (#22959)
+* Deprecate params `subnet` and `vnet-name` on `network-rule` command group (#22959)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:21]:
```suggestion
* `az acr network-rule`: Deprecate params `--subnet` and `--vnet-name` (#22959)
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,150 @@
...
+* Introducing securitySolutionsReferenceData CLI (Updated) (#23456)
+
+**Service Bus**
+
+* `az servicebus namespace create/update`: Support specifying `--min-tls` (#23697)
+* `az servicebus namespace network-rule update`: Support updating network rules for given namespace (#23697)
+
+**Service Connector**
+
+* Update `az spring connection` description after spring app rename (#23616)
```
Comment (wangzelin007) [inline src/azure-cli/HISTORY.rst:128]:
```suggestion
* `az spring connection`: Update description after spring app rename (#23616)
```

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,150 @@
...
+* `az acr config authentication-as-arm update`: Add new command to support updating 'Azure AD authenticate as ARM' policy (#23323)
+* `az acr manifest show`: Improve performance (#23523)
+* `az acr config soft-delete show`: Add new command to show soft-delete policy (#22959)
+* `az acr config soft-delete update`: Add new command to update soft-delete policy (#22959)
+* `az acr repository list-deleted`: Add new command to list deleted repositories (#22959)
+* `az acr manifest list-deleted`: Add new command to list deleted manifests (#22959)
+* `az acr manifest list-deleted-tags`: Add new command to list deleted tags (#22959)
+* `az acr manifest restore`: Add new command to restore deleted manifests and tags (#22959)
+* Deprecate params `subnet` and `vnet-name` on `network-rule` command group (#22959)
+* Fix bug in some `acr config` commands that would in certain circumstances attempt to pull a nonexistent model from SDK (#22959)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:22]:
```suggestion
* `acr config`: Fix bug in some commands that would in certain circumstances attempt to pull a nonexistent model from SDK (#22959)
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,150 @@
...
+* `az monitor metric alert`: Support metric namespace with dash (#23637)
+* `az monitor action-group create`: Add optional `--location` parameter to `create` cmd (#23619)
+
+**NetAppFiles**
+
+* `az netappfiles volume create`: Add optional parameter `--kv_private_endpoint_id` (#23439)
+* `az netappfiles volume-group create`: Add optional parameter `--kv_private_endpoint_id` (#23439)
+* `az netappfiles volume` update: Add optional parameter `--cool_access` (#23439)
+* `az netappfiles volume` update: Add optional parameter `--coolness_period` (#23439)
+* `az netappfiles pool` update: Add optional parameter `--cool_access` (#23439)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:93]:
```suggestion
* `az netappfiles volume update`: Add optional parameter `--cool_access` (#23439)
* `az netappfiles volume update`: Add optional parameter `--coolness_period` (#23439)
* `az netappfiles pool update`: Add optional parameter `--cool_access` (#23439)
```

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,150 @@
...
+* `az webapp up`: Show status during deployment for linux apps (#23464)
+* `az webapp deployment source config-zip`: Show status during deployment for linux apps (#23464)
+* `az logicapp deployment`: Add a new command group to support managing logic app deployments (#23585)
+* `az logicapp scale`: Add a new command to support scaling a logic app (#23585)
+* `az logicapp config`: Add a new command group to support configuring a logic app (#23585)
+* `az logicapp update`: Add a new command to support updating a logic app (#23585)
+
+**ARM**
+
+* `azure bicep`: Use `AZURE_CLI_DISABLE_CONNECTION_VERIFICATION` when checking Bicep CLI versions (#23524)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:49]:
```suggestion
* `az bicep`: Use `AZURE_CLI_DISABLE_CONNECTION_VERIFICATION` when checking Bicep CLI versions (#23524)
```

File: src/azure-cli/HISTORY.rst [hunk 10]
```diff
@@ -3,6 +3,148 @@
 Release History
 ===============
 
+2.40.0
+++++++
+
+**ACR**
+* `az acr config authentication-as-arm show`: Add new command to support showing the configured 'Azure AD authenticate as ARM' policy (#23323)
+* `az acr config authentication-as-arm update`: Add new command to support updating 'Azure AD authenticate as ARM' policy (#23323)
+* `az acr manifest show`: Improve performance (#23523)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:12]:
```suggestion
```

File: src/azure-cli/HISTORY.rst [hunk 11]
```diff
@@ -3,6 +3,150 @@
...
+* `az netappfiles volume create`: Add optional parameter `--kv_private_endpoint_id` (#23439)
+* `az netappfiles volume-group create`: Add optional parameter `--kv_private_endpoint_id` (#23439)
+* `az netappfiles volume` update: Add optional parameter `--cool_access` (#23439)
+* `az netappfiles volume` update: Add optional parameter `--coolness_period` (#23439)
+* `az netappfiles pool` update: Add optional parameter `--cool_access` (#23439)
+
+**Network**
+
+* `az network application-gateway create`: Support rule priority field provided as part of configuration (#23438)
+* Private link add `Microsoft.OpenEnergyPlatform/energyServices` provider (#23587)
```
Comment (necusjz) [inline src/azure-cli/HISTORY.rst:96]:
```suggestion
* `az network private-endpoint-connection`: Add `Microsoft.OpenEnergyPlatform/energyServices` provider (#23587)
```

File: src/azure-cli/HISTORY.rst [hunk 12]
```diff
@@ -3,6 +3,150 @@
...
+
+**Feedback**
+
+* `az survey`: New command for CLI survey (#23460)
+
+**Monitor**
+
+* Add new command: `az monitor action-group test-notifications create` (#23411)
+* `az monitor metric alert`: Support metric namespace with dash (#23637)
+* `az monitor action-group create`: Add optional `--location` parameter to `create` cmd (#23619)
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:83]:
```suggestion
* `az monitor action-group create`: Add optional `--location` parameter (#23619)
```

**Comment by yonzhan** [comment]
Release

File: src/azure-cli/HISTORY.rst [hunk 13]
```diff
@@ -3,6 +3,150 @@
...
+* `az mysql/postgres flexible-server backup`: Add backup commands for flexible servers (#23432)
+* `az postgres flexible-server create/update`: Add `SameZone` for HA in PostgreSQL flexible server (#23473)
+* `az mysql flexible-server create/update/restore/geo-restore/replica`: Add BYOK for MySQL Flexible Server (#23197)
+* `az mysql flexible-server identity/ad-admin`: User Identity and AAD Admin for MySQL flexible server (#23474)
+
+**Security**
+
+* Introducing securitySolutionsReferenceData CLI (#23336)
+* Introducing securitySolutions CLI (#23361)
+* Introducing securitySolutionsReferenceData CLI (Updated) (#23456)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:117]:
```suggestion
* `az security security-solutions-reference-data`: Add new command group (#23336)
* `az security security-solutions`: Add new command group (#23361)
```

File: src/azure-cli/HISTORY.rst [hunk 14]
```diff
@@ -3,6 +3,148 @@
 Release History
 ===============
 
+2.40.0
+++++++
+
+**ACR**
+* `az acr config authentication-as-arm show`: Add new command to support showing the configured 'Azure AD authenticate as ARM' policy (#23323)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:10]:
```suggestion

* `az acr config authentication-as-arm show`: Add new command to support showing the configured 'Azure AD authenticate as ARM' policy (#23323)
```

**Review by necusjz — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli-core/HISTORY.rst [hunk 15]
```diff
@@ -3,6 +3,15 @@
 Release History
 ===============
 
+2.40.0
+++++++
+* `aaz`: Support using `cli_ctx` to initiate AAZCommand class (#23459)
+* `aaz`: Fix AAZBearerTokenCredentialPolicy to support MSIAuthenticationWrapper credential (#23471)
+* Allow adding dedicated key for collecting cli telemetry (#23530)
```
Comment (zhoxing-ms) [inline src/azure-cli-core/HISTORY.rst:10]:
```suggestion
* Allow adding dedicated key for collecting CLI Telemetry (#23530)
```

File: src/azure-cli/HISTORY.rst [hunk 16]
```diff
@@ -3,6 +3,147 @@
...
+**Monitor**
+
+* `az monitor action-group test-notifications create`: Add new command (#23411)
+* `az monitor metric alert`: Support metric namespace with dash (#23637)
+* `az monitor action-group create`: Add optional parameter `--location` (#23619)
+
+**NetAppFiles**
+
+* `az netappfiles volume create`: Add optional parameter `--kv_private_endpoint_id` (#23439)
+* `az netappfiles volume-group create`: Add optional parameter `--kv_private_endpoint_id` (#23439)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:88]:
```suggestion
* `az netappfiles volume create`: Add optional parameter `--kv-private-endpoint-id` (#23439)
* `az netappfiles volume-group create`: Add optional parameter `--kv-private-endpoint-id` (#23439)
```

File: src/azure-cli/HISTORY.rst [hunk 17]
```diff
@@ -3,6 +3,147 @@
...
+* `az netappfiles volume update`: Add optional parameter `--cool_access` (#23439)
+* `az netappfiles volume update`: Add optional parameter `--coolness_period` (#23439)
+* `az netappfiles pool update`: Add optional parameter `--cool_access` (#23439)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:91]:
```suggestion
* `az netappfiles volume create`: Add optional parameter `--kv-private-endpoint-id` (#23439)
* `az netappfiles volume-group create`: Add optional parameter `--kv-private-endpoint-id` (#23439)
* `az netappfiles volume update`: Add optional parameter `--cool-access` (#23439)
* `az netappfiles volume update`: Add optional parameter `--coolness-period` (#23439)
* `az netappfiles pool update`: Add optional parameter `--cool-access` (#23439)
```

**Pull request merged**